Tasks/VsTest/distributedtest.ts

```diff
@@ -0,0 +1,137 @@
+import * as fs from 'fs';
+import * as path from 'path';
+import * as Q from 'q';
+import * as ps from 'child_process';
+import * as tl from 'vsts-task-lib/task';
+import * as tr from 'vsts-task-lib/toolrunner';
+import * as models from './models';
+import * as settingsHelper from './settingsHelper';
+import * as utils from './helpers';
+import * as ta from './testAgent';
+import * as versionFinder from './versionFinder';
+
+export class DistributedTest {
+    constructor(dtaTestConfig: models.DtaTestConfigurations) {
+        this.dtaPid = -1;
+        this.dtaTestConfig = dtaTestConfig;
+    }
+
+    public runDistributedTest() {
+        this.registerAndConfigureAgent();
+    }
+
+   private async registerAndConfigureAgent() {
+        tl.debug('Configure the Agent with DTA... Invoking the createAgent REST API');
+
+        try {
+            const agentId = await ta.TestAgent.createAgent(this.dtaTestConfig.dtaEnvironment, 3);
+            this.dtaPid =  await this.startDtaExecutionHost(agentId);
+            await this.startDtaTestRun();
+            try {
+                // TODO: ranjanar: fixasap : This dangerous - Note that the PID only uniquely identifies your child process 
+                // as long as it is running. After it has exited, the PID might have been reused for a different process.
+                process.kill(this.dtaPid);
```

Comment:
remove TODO and check for process name before killing and to eliminate the possibility of killing some random process